### PR TITLE
Dialog flashing bug fix

### PR DESCRIPTION
**Pull request by Astro-Alien**

### CODE REVIEW

File: app/dialog/dialog.js [hunk 1]
```diff
@@ -37,6 +41,33 @@ export default class Dialog extends crsbinding.classes.ViewBase {
         });
     }
 
+    async showNestedDialog(event) {
+        const instance = this._element.querySelector("#nested-dialog").content.cloneNode(true);
+        const position = document.querySelector("#positionOption").value;
+
+        const anchor = {
```
Comment (Kieran-Reilly) [inline app/dialog/dialog.js]: Break into a private property #anchor and use that here. Update the rest accordingly

File: app/dialog/dialog.js [hunk 2]
```diff
@@ -37,6 +41,33 @@ export default class Dialog extends crsbinding.classes.ViewBase {
         });
     }
 
+    async showNestedDialog(event) {
```
Comment (Kieran-Reilly) [inline app/dialog/dialog.js]: Double check we're implementing the nested scenario properly here as per discussed

File: app/dialog/dialog.js (hunk 1 above)
Comment (Astro-Alien) [reply app/dialog/dialog.js]: Created #anchor private property and implemented the changes where it was required.

File: app/dialog/dialog.js [hunk 3]
```diff
@@ -20,23 +34,109 @@ export default class Dialog extends crsbinding.classes.ViewBase {
...
+                    await this.#showChildDialog();
+                }
+            }
+        });
+    }
+
+    async #showChildDialog() {
+        const instance = this._element.querySelector("#child-dialog").content.cloneNode(true);
+
+        const header = await crs.call("dom", "create_element", {
```
Comment (Kieran-Reilly) [inline app/dialog/dialog.js]: This looks like a copy-paste of the above header you're creating for the parent dialog. In this situation, please add a template on the html with a relevant id, and just retrieve the template and clone it's content for both here and the parent dialog - similar to what we're doing on line 99 above

File: components/dialog/dialog.js [hunk 4]
```diff
@@ -290,7 +298,12 @@ export class Dialog extends HTMLElement {
         const popup = this.shadowRoot.querySelector(".popup");
 
         if (options?.target == null) {
-            return await crs.call("fixed_position", "set", {element: popup, container: options?.parent, position: "center-screen", margin: 10});
+            return await crs.call("fixed_position", "set", {
```
Comment (Kieran-Reilly) [inline components/dialog/dialog.js]: Not necessary to have all these properties on separate lines

File: components/dialog/dialog.js [hunk 5]
```diff
@@ -332,7 +346,15 @@ export class Dialog extends HTMLElement {
     async show(header, main, footer, options) {
         const struct = {header, main, footer, options};
         this.#stack.push(struct);
-        await this.#showStruct(struct);
+        this.style.opacity = 0;
+
+        if (this.#stack.length > 1) {
```
Comment (Kieran-Reilly) [inline components/dialog/dialog.js]: I don't see why we have this if-else. You call the same function in both, but on the if you wrap the #showStruct function call in a setTimeout? Please have a look at this

File: components/dialog/dialog.js [hunk 6]
```diff
@@ -252,19 +258,21 @@ export class Dialog extends HTMLElement {
 
         let bodyElement = this.querySelector("[slot=body]");
 
-        if(bodyElement == null) {
+        if (bodyElement == null) {
             bodyElement = document.createElement("div");
             bodyElement.setAttribute("slot", "body");
             this.appendChild(bodyElement);
+        } else {
+            bodyElement.textContent = "";
```
Comment (Kieran-Reilly) [inline components/dialog/dialog.js]: Let's maybe chat on this. I want to understand why we need to set the textContent on the bodyElement to empty string if the bodyElement is already defined

**Review by Kieran-Reilly — CHANGES_REQUESTED** [review]

File: app/dialog/dialog.js (hunk 2 above)
Comment (Astro-Alien) [reply app/dialog/dialog.js]: checked the scenarios and implemented a few changes to resolve issues encountered.

File: app/dialog/dialog.js (hunk 3 above)
Comment (Astro-Alien) [reply app/dialog/dialog.js]: created the template and replaced the duplicate code.

File: components/dialog/dialog.js (hunk 5 above)
Comment (Astro-Alien) [reply components/dialog/dialog.js]: removed the if else and made the adjustment necessary for the test to run without time delay.

File: components/dialog/dialog.js (hunk 4 above)
Comment (Astro-Alien) [reply components/dialog/dialog.js]: set all properties back to its previous line.

File: components/dialog/dialog.js (hunk 6 above)
Comment (Astro-Alien) [reply components/dialog/dialog.js]: replaced .textContent with .replaceChildren to prevent parent dialog data appending to child dialog data.

**Review by Kieran-Reilly — APPROVED** [review]

**Pull request merged**